lib/core/src/Cardano/Wallet/DB/Sqlite.hs

```diff
@@ -281,6 +296,36 @@ newDBFactory tr defaultFieldValues mDatabaseDir = do
                 databaseFilePrefix <> "." <>
                 T.unpack (toText wid) <> ".sqlite"
 
+internalRemovePathForcibly :: FilePath -> IO ()
+internalRemovePathForcibly =
+#if defined(mingw32_HOST_OS)
+    -- On Windows we have to retry the delete a couple of times.
+    -- The reason for this is that a FileDelete command just marks a
+    -- file for deletion. The file is really only removed when the last
+    -- handle to the file is closed. Unfortunately there are a lot of
+    -- system services that can have a file temporarily opened using a shared
+    -- read-only lock, such as the built in AV and search indexer.
+    --
+    -- We can't really guarantee that these are all off, so what we can do is
+    -- whenever after an rm the file still exists to try again and wait a bit.
+    retryOnPermissionErrors 5 . removePathForcibly
+#else
+    removePathForcibly
+#endif
+
+#if defined(mingw32_HOST_OS)
+retryOnPermissionErrors :: Int -> IO a -> IO a
+retryOnPermissionErrors maxRetries io = do
+    res <- try io
+    case res of
+        Right a -> return a
+        Left ex | isPermissionError ex && maxRetries > 1 -> do
+                    let retries' = traceShow (ex, maxRetries) (maxRetries - 1)
+                    threadDelay ((maxRetries - retries') * 200)
+                    retryOnPermissionErrors retries' io
+                | otherwise -> traceShow ("OTHER ERROR" :: String) (throw ex)
```

Comment:
If we end up using the retry approach: You can log with the iohk-monitoring tracer here, rather than Debug.Trace.traceShow. Also some debug logging on retries would help.

Reply:
Yes, though is this falling outside my haskell proficiency atm :(.
There is debug logging on retries at line 323:
```haskell
 let retries' = traceShow (ex, maxRetries) (maxRetries - 1)
```

Reply:
OK - I will convert the logging to use Control.Tracer.